distribution/test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/TestSingleSignOn.java

```diff
@@ -0,0 +1,261 @@
+
+    public static final String IDP_AUTH_TYPES = "/=SAML|ANON,/search=SAML|IDP,/solr=SAML|PKI|basic";
+
+    private static final DynamicUrl SEARCH_URL = new DynamicUrl(DynamicUrl.SECURE_ROOT, HTTPS_PORT,
+            "/search");
+
+    private static final DynamicUrl WHO_AM_I_URL = new DynamicUrl(SERVICE_ROOT, "/whoami");
+
+    @BeforeExam
+    public void beforeTest() throws Exception {
+        basePort = getBasePort();
+        getAdminConfig().setLogLevels();
+        getSecurityPolicy().configureWebContextPolicy(null, IDP_AUTH_TYPES, null, null);
+        getServiceManager().waitForAllBundles();
+        getServiceManager().waitForHttpEndpoint(SERVICE_ROOT + "/catalog/query");
+        getServiceManager().waitForHttpEndpoint(WHO_AM_I_URL.getUrl());
+
+        // Start the services needed for testing. We need to start the Search UI to test that it redirects properly
+        getServiceManager().startFeature(true, "security-idp", "search-ui-app");
+
+        // Create a validator from the metadata schema so that we can validate the metadata
+        // TODO: Maybe don't go out to the internet for this? The schema itself is going to go out to the internet anyway though.
+        URL schemaFile = new URL(
+                "http://docs.oasis-open.org/security/saml/v2.0/saml-schema-metadata-2.0.xsd");
+        SchemaFactory schemaFactory = SchemaFactory.newInstance(XMLConstants.W3C_XML_SCHEMA_NS_URI);
+        Schema schema = schemaFactory.newSchema(schemaFile);
+        Validator validator = schema.newValidator();
+
+        // Get the metadata
+        String clientMetadata = get(SERVICE_ROOT + "/idp/login/metadata").asString();
+        String serverMetadata = get(SERVICE_ROOT + "/saml/sso/metadata").asString();
+
+        // TODO: The server metadata requires an "index" attribute to validate properly
+        validator.validate(new StreamSource(new StringReader(clientMetadata)));
+        //        validator.validate(new StreamSource(new StringReader(serverMetadata)));
+
+        // To find the right inputs for the settings, go into the metatype.xml file.
+        // The key is the "id" and the value type is determined by the cardinality as such:
+        // Positive = array, negative = vector, 0 (none) = single variable
+        Dictionary<String, Object> clientSettings = new Hashtable<>();
+        clientSettings.put("metadata", clientMetadata);
+        Dictionary<String, Object> serverSettings = new Hashtable<>();
+        serverSettings.put("spMetadata", new String[] {serverMetadata});
+
+        // Update the client and server with the metadata
+        Configuration clientConfig = getAdminConfig()
+                .getConfiguration("org.codice.ddf.security.idp.client.IdpMetadata", null);
+        Configuration serverConfig = getAdminConfig()
+                .getConfiguration("org.codice.ddf.security.idp.server.IdpEndpoint", null);
+        clientConfig.update(clientSettings);
+        serverConfig.update(serverSettings);
```

Comment:
Agreed. I just looked for a quick example to include and didn't see a config change using `WaitCondition`. I should have spent more than two minutes on that.